### ISSUE TITLE
Relative to the old DES_key_schedule, what is 'key' for EVP_EncryptInit_ex() and DES-ECB?

### ISSUE THREAD

**Issue by jjeffroberts** [openssl/openssl#20419]
I'm working through upgrades to 3.0 and we have some legacy code that uses the deprecated DES_ecb_encrypt().

When I replace that with code that includes calling EVP_EncryptInit_ex():

```
int EVP_EncryptInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
                       ENGINE *impl, const unsigned char *key, const unsigned char *iv);
```

1. Relative to the DES_key_schedule that was passed to DES_ecb_encrypt(), what should I pass as **key** for DES-ECB ?

The existing code:

- Starts with a key I'll call k1 as an unsigned char array
- Sends k1 to a method that:
   -  Per comment "turns a 56 bit key into the 64 bit key" I'll call k2
   -  Calls [deprecated] DES_set_odd_parity on k2, I'll call the result k3
   - Passes k3 to DES_set_key() to set it into a DES_key_schedule ks.

2. Am I right that **iv** is NULL for DES-ECB?

So far I've loaded the legacy provider and tried key=k1 and iv=NULL, but don't get the same result as the original code.

I'd need to refactor slightly to try k2, etc.

It would be helpful to describe what **key** should be in the man page if it is not k1.

If it turns out that k1 should work I can create a simpler example.

Thanks,
-Jeff

**Comment by slontis** [comment]
Maybe have a look at some test vectors that use the same inputs to figure this out.. 
i.e. test/destest.c uses the DES API

Which looks like it runs same tests as this
````
LD_LIBRARY_PATH=. OPENSSL_MODULES=providers ./test/evp_test -config test/default-and-legacy.cnf test/recipies/30-test_evp_data/evp_ciph_des.txt
````
Which uses the EVP layer.

**Comment by jjeffroberts** [comment]
The answer appears to be k2 - I need to keep "our" function that grabs 56 bits of key material and turns it into a 64-bit key.

Looking at the output of that function, it doesn't seem to be fixing up the parity, and I'm no longer calling the deprecated DES_set_odd_parity(), so the parity of the key passed to EVP_EncryptInit_ex() must not matter.

It might be helpful if the docs were clear that it's not necessary to fix up the parity for the cases where it used to matter.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
